File: CSharp/7kyu/CreditCardMask.cs
using System.Linq;

namespace CSharp._7Kyu
{
    public static partial class Kata7
    {
        public static string Maskify(string cc)
        {
            int len = cc.Length;
            return len < 4
                ? cc
                : new string('*', len - 4) + cc.Substring(len - 4);
        }

        public static string Maskify_2(string cc)
        {
            int len = cc.Length;
            if (len < 4)
            {
                return cc;
            }

            return cc.Substring(len - 4).PadLeft(len, '#');
        }
    }
}

File: JavaScript/7kyu/CreditCardMask.js
function maskify(cc) {
  const len = cc.length - 4;

  if (len <= 0) {
    return cc;
  }

  return "#".repeat(len) + cc.substr(len);
}

function maskify_2(cc) {
  return cc.replace(/\d(?=....)/g, "#");
}

var result = maskify_2("11111231231231");
console.log(result);
